docs/source/app-dev/services.rst

```diff
@@ -59,6 +59,23 @@ Commands can be labeled with two application-specific IDs, both of which are ret
 
 For full details, see :ref:`the proto documentation for the service <com.digitalasset.ledger.api.v1.CommandSubmissionService>`.
 
+.. _command-submission-service-deduplication:
+
+Command deduplication
+---------------------
+
+The command submission service deduplicates submitted commands:
+
+- Applications can assign a "time to live" (TTL) value to each command.
+- Duplicate command submissions (using the same submitter and command ID) will be ignored until either the TTL of the original command has elapsed or the original submission was rejected, whichever comes first.
+- Command deduplication is only guaranteed to work if all commands are submitted to the same participant. On some ledgers, the deduplication may also work across participants.
+- A command submission will return:
+  - The result of the submission, if the command was submitted for the first time
+  - The status error ALREADY_EXISTS, if the command was deduplicated and the result of the first submission is unknown
+  - The result of the original submission, if the command was deduplicated and the result of the original submission is known
```

Comment:
Both command submission service and command service's SubmitAndWait return `google.protobuf.Empty` on success. I think it is misleading to talk about "result" here, when it's usually an empty message (the exception being the other methods in `CommandService`).

Reply:
We've been postponing changing this due to it being a breaking change. It never should've been just `Empty`. Maybe we change it for DAML 1.0?

Reply:
I see that it's misleading. As noted above, this is talking about returning either `google.protobuf.Empty` if the original submission was successful or the same gRPC error as the original submission if the original submission failed.